### PR TITLE
Minor fixes and improvements

### PR DESCRIPTION
**Pull request by nicovogelaar**
# Description

Please provide a detailed description of what was done in this PR

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: blockchain/blockchain.go [hunk 1]
```diff
@@ -89,37 +89,32 @@ func NewBlockchain(
...
-	b.db = storage
+	b.db = db
 
 	b.headersCache, _ = lru.New(100)
 	b.difficultyCache, _ = lru.New(100)
 
 	// Push the initial event to the stream
 	b.stream.push(&Event{})
 
-	if _, ok := consensus.(*MockVerifier); ok {
```
Comment (nicovogelaar) [inline blockchain/blockchain.go:115]: Testing code should not be part of the actual implementation.

File: blockchain/subscription.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"github.com/0xPolygon/minimal/types"
 )
 
+type void struct{}
```
Comment (nicovogelaar) [inline blockchain/subscription.go:10]: for readability (optional)

File: blockchain/subscription.go [hunk 3]
```diff
@@ -55,6 +57,9 @@ func (s *subscription) GetEventCh() chan *Event {
 	go func() {
 		for {
 			evnt := s.GetEvent()
+			if evnt == nil {
```
Comment (nicovogelaar) [inline blockchain/subscription.go:60]: `GetEvent()` will return `nil` when the subscription is closed.

File: blockchain/blockchain.go [hunk 4]
```diff
@@ -89,37 +89,32 @@ func NewBlockchain(
 		stream:    &eventStream{},
 	}
 
-	var storage storage.Storage
```
Comment (nicovogelaar) [inline blockchain/blockchain.go:92]: Avoid package/variable name collisions.

**Review by gatorbate32038 — APPROVED** [review]

**Review by Vuksan — APPROVED** [review]
Thank you for your contribution! 🙏

**Review by zivkovicmilos — APPROVED** [review]
Looks alright, just need to resolve merge conflicts if they arise since this is an older PR 👍

**Pull request merged**